evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -1024,3 +1076,129 @@ def test_partial_dependence_respect_grid_resolution(fraud_100):
+    assert fig_dict["data"][0]["name"] == expected_label
+    assert fig_dict["data"][-1]["name"] == "Partial Dependence" if problem_type == ProblemTypes.BINARY else "Partial Dependence: class_2"
+
+    avg_dep_data, ind_dep_data = partial_dependence(
+        clf, X, features=feature, grid_resolution=20, kind="both"
+    )
+    assert np.array_equal(fig_dict["data"][-1]["x"], avg_dep_data["feature_values"][:len(fig_dict["data"][-1]["x"])].values)
+
+    if problem_type == ProblemTypes.BINARY:
+        assert np.array_equal(
+            fig_dict["data"][-1]["y"], avg_dep_data["partial_dependence"].values
+        )
+    else:
+        class_2_data = avg_dep_data[avg_dep_data['class_label'] == 'class_2']["partial_dependence"].values
+        assert np.array_equal(
+            fig_dict["data"][-1]["y"], class_2_data
+        )
+
+    for i in range(len(X)):
+        assert np.array_equal(fig_dict["data"][i]["x"], ind_dep_data["feature_values"])
+        if problem_type == ProblemTypes.MULTICLASS:
+            window_length = len(ind_dep_data[f"Sample {i}"].values)
+            data = np.concatenate([fig_dict["data"][i]["y"][:window_length],
+                                   fig_dict["data"][i + len(X) + 1]["y"][:window_length],
+                                   fig_dict["data"][i + 2 * len(X) + 2]["y"][:window_length]])
+            assert np.array_equal(
+                data, ind_dep_data[f"Sample {i}"].values
+            )
+        else:
+            assert np.array_equal(
+                fig_dict["data"][i]["y"], ind_dep_data[f"Sample {i}"].values
+            )
+
+    fig = graph_partial_dependence(
+        clf, X, features=feature, grid_resolution=20, kind="individual"
+    )
+    assert isinstance(fig, go.Figure)
+    fig_dict = fig.to_dict()
+    assert (
+        fig_dict["layout"]["title"]["text"]
+        == f"Individual Conditional Expectation of '{feature}'"
+    )
+    assert len(fig_dict["data"]) == len(X) * (len(y.unique()) if problem_type == ProblemTypes.MULTICLASS else 1)
+    expected_label = "Individual Conditional Expectation" if problem_type == ProblemTypes.BINARY else "Individual Conditional Expectation: class_0"
+    assert fig_dict["data"][0]["name"] == expected_label
+
+    ind_dep_data = partial_dependence(
+        clf, X, features=feature, grid_resolution=20, kind="individual"
+    )
+
+    for i in range(len(X)):
+        assert np.array_equal(fig_dict["data"][i]["x"], ind_dep_data["feature_values"])
+        if problem_type == ProblemTypes.MULTICLASS:
+            window_length = len(ind_dep_data[f"Sample {i}"].values)
+            data = np.concatenate([fig_dict["data"][i]["y"][:window_length],
+                                   fig_dict["data"][i + len(X)]["y"][:window_length],
+                                   fig_dict["data"][i + 2 * len(X)]["y"][:window_length]])
```

Comment:
Same as the other comment, but the +1 and +2 aren't here since we're only plotting the ICE data.